### DISCUSSION TITLE
Unable to load shared library 'ldap.so.2'

### DISCUSSION THREAD

**Discussion by OleksandrPBolt** [flamencist/ldap4net#168] [Q&A]
Hi.
I'm trying to run a .NET 5 WebApi that uses ldap4net in a Docker container that is running standard ASP.NET 5 runtime image.
However ldap is not found, even after installing it and copying files to /usr/lib.
Any ideas?

My Dockerfile:

`FROM mcr.microsoft.com/dotnet/sdk:5.0 AS build-env
WORKDIR /app

WORKDIR /src
COPY . ./
RUN dotnet restore "Appetite.csproj" -s "https://api.nuget.org/v3/index.json"
WORKDIR "/src/Appetite"
RUN dotnet publish -c Debug -o /app/out

FROM mcr.microsoft.com/dotnet/aspnet:5.0
WORKDIR /app

RUN apt-get update \
    && apt-get install -y --no-install-recommends libldap-2.4-2 \
    && apt-get install -y --no-install-recommends libldap-common \
    && rm -rf /var/lib/apt/lists/* 

RUN ln -s /usr/lib/x86_64-linux-gnu/libldap-2.4.so.2.11.5 /usr/lib/libldap.so.2
RUN ln -s /usr/lib/x86_64-linux-gnu/liblber_r-2.4.so.2.11.5 /usr/lib/liblber.so.2

COPY --from=build-env /app/out .
EXPOSE 5000
EXPOSE 5001
ENTRYPOINT ["dotnet", "Appetite.dll"]`

**Comment by Kashemir78** [comment]
Hi. 
I hope it can help. My docker (aspnet:6.0 and the same for aspnet:3.1 ) works ok:

FROM mcr.microsoft.com/dotnet/aspnet:6.0 AS base
WORKDIR /app
EXPOSE 80
EXPOSE 443


RUN apt-get update \
    && apt-get install -y --no-install-recommends libldap-2.4-2 \
    && apt-get install -y --no-install-recommends libldap-common \
    && rm -rf /var/lib/apt/lists/* \
    && ln -s -f /usr/lib/x86_64-linux-gnu/libldap-2.4.so.2 /usr/lib/x86_64-linux-gnu/libldap.so.2 \
    && ln -s -f /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2 /usr/lib/x86_64-linux-gnu/liblber.so.2


COPY ["ActiveDirectoryWindowsAuth/SSL", "/usr/local/share/ca-certificates"]
RUN update-ca-certificates


FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build
WORKDIR /src
COPY ["ActiveDirectoryWindowsAuth/ActiveDirectoryWindowsAuth.csproj", "ActiveDirectoryWindowsAuth/"]
RUN dotnet restore "ActiveDirectoryWindowsAuth/ActiveDirectoryWindowsAuth.csproj"
COPY . .
WORKDIR "/src/ActiveDirectoryWindowsAuth"
RUN dotnet build "ActiveDirectoryWindowsAuth.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "ActiveDirectoryWindowsAuth.csproj" -c Release -o /app/publish

FROM base AS final


WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "ActiveDirectoryWindowsAuth.dll"]
